diff -ru orig/8cj72bcgzy2s733l0nz5g4dgxsz9vrrk-emacs-gptel-0.9.5-checkout/gptel-rewrite.el 8cj72bcgzy2s733l0nz5g4dgxsz9vrrk-emacs-gptel-0.9.5-checkout/gptel-rewrite.el
--- orig/8cj72bcgzy2s733l0nz5g4dgxsz9vrrk-emacs-gptel-0.9.5-checkout/gptel-rewrite.el	2024-11-29 17:46:41.209088124 +0100
+++ 8cj72bcgzy2s733l0nz5g4dgxsz9vrrk-emacs-gptel-0.9.5-checkout/gptel-rewrite.el	2024-11-29 17:55:17.903792163 +0100
@@ -104,7 +104,7 @@
       (funcall callback
                (format (substitute-command-keys "%s rewrite available: accept \\[gptel--rewrite-apply], clear \\[gptel--rewrite-clear], merge \\[gptel--rewrite-merge], diff \\[gptel--rewrite-diff] or ediff \\[gptel--rewrite-ediff]")
                        (propertize (concat (gptel-backend-name gptel-backend)
-                                           ":" gptel-model)
+                                           ":" (gptel--model-name gptel-model))
                                    'face 'mode-line-emphasis)))))
 
 (defun gptel--rewrite-move (search-func)
@@ -354,7 +354,7 @@
                       (setq response (replace-regexp-in-string "^```.*$" "" response))))
                 (setq action-str "rewrite"))
               (setq hint-str (concat "[" (gptel-backend-name gptel-backend)
-                                     ":" gptel-model "] " (upcase action-str)
+                                     ":" (gptel--model-name gptel-model) "] " (upcase action-str)
                                      " READY ✓\n"))
               (add-hook 'eldoc-documentation-functions #'gptel--rewrite-key-help nil 'local)
               (overlay-put ov 'gptel-rewrite response)
@@ -372,7 +372,7 @@
 - diff   \\[gptel--rewrite-diff],
 - ediff  \\[gptel--rewrite-ediff]"
                                    (propertize (concat (gptel-backend-name gptel-backend)
-                                                       ":" gptel-model))))
+                                                       ":" (gptel--model-name gptel-model)))))
               (push ov gptel--rewrite-overlays))
             ;; Message user
             (message
