diff -ru orig/imaplib2-3.6/imaplib2/imaplib2.py imaplib2-3.6/imaplib2/imaplib2.py
--- orig/imaplib2-3.6/imaplib2/imaplib2.py	2021-06-10 22:38:36.000000000 +0200
+++ imaplib2-3.6/imaplib2/imaplib2.py	2025-01-24 22:42:48.561429102 +0100
@@ -509,7 +509,7 @@
                 raise RuntimeError("unknown tls_level: %s" % self.tls_level)
 
             if self.ssl_version not in TLS_MAP[self.tls_level]:
-                raise socket.sslerror("Invalid SSL version '%s' requested for tls_version '%s'" % (self.ssl_version, self.tls_level))
+                raise ssl.SSLError("Invalid SSL version '%s' requested for tls_version '%s'" % (self.ssl_version, self.tls_level))
 
             ssl_version =  TLS_MAP[self.tls_level][self.ssl_version]
 
@@ -528,7 +528,7 @@
             self.read_fd = self.sock.fileno()
         except ImportError:
             # No ssl module, and socket.ssl has no fileno(), and does not allow certificate verification
-            raise socket.sslerror("imaplib SSL mode does not work without ssl module")
+            raise ssl.SSLError("imaplib SSL mode does not work without ssl module")
 
         if self.cert_verify_cb is not None:
             cert_err = self.cert_verify_cb(self.sock.getpeercert(), self.host)
