--- a/yt_dlp/extractor/xhamster.py
+++ b/yt_dlp/extractor/xhamster.py
@@ -1,5 +1,6 @@
 import itertools
 import re
+import base64
 
 from .common import InfoExtractor
 from ..utils import (
@@ -229,6 +230,11 @@ class XHamsterIE(InfoExtractor):
                                 standard_url = standard_format.get(standard_format_key)
                                 if not standard_url:
                                     continue
+                                if standard_url.startswith("eG9yXxAcQ0"): # base64 of 'xor'
+                                    xor_url_content=base64.b64decode(standard_url)[4:]
+                                    standard_url = "";
+                                    for t in range( 0, len(xor_url_content) ):
+                                        standard_url += chr( xor_url_content[t] ^ b"xh7999"[ t % 6 ] );
                                 standard_url = urljoin(url, standard_url)
                                 if not standard_url or standard_url in format_urls:
                                     continue